src/background/services/sendToPopup.ts

```diff
@@ -0,0 +1,52 @@
+import type { AmountValue, Storage } from '@/shared/types'
+import type { Browser, Runtime } from 'webextension-polyfill'
+
+interface BackgroundToPopupMessages {
+  SET_BALANCE: Record<'recurring' | 'oneTime' | 'total', AmountValue>
+  SET_STATE: { state: Storage['state']; prevState: Storage['state'] }
+}
+
+export type BackgroundToPopupMessage = {
+  [K in keyof BackgroundToPopupMessages]: {
+    type: K
+    data: BackgroundToPopupMessages[K]
+  }
+}[keyof BackgroundToPopupMessages]
+
+export const CONNECTION_NAME = 'popup'
```

Comment:
I was importing them from background into popup. I think that keeps "source" clearer.

Reply:
Note that this is only background -> popup, and not popup -> background (we do that with regular `runtime.onMessage` API).

Reply:
I wouldn't worry about "importing" background into popup, as bundler will remove all the unused bits.

Reply:
Sorry, my bad. Not the "shared types folder". For the popup -> background scenario, the messages are defined in `shared/messages`. Let's do the same for background -> popup.

Not worrying about importing background into popup, but for consistency, since we already have this approach for defining the messages between the extension components.